parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -434,6 +440,63 @@ public static final ParquetMetadata readFooter(Configuration configuration, File
     }
   }
 
+  public static final Map<ColumnDescriptor, Dictionary> readDictionary(BlockMetaData blockMetadata,
+                                                                        Path path,
+                                                                        Configuration configuration,
+                                                                        MessageType requestedSchema) throws IOException
+  {
+    Map<ColumnDescriptor, Dictionary> dictionaries = new HashMap<ColumnDescriptor, Dictionary>();
+    for (ColumnChunkMetaData columnChunkMetaData : blockMetadata.getColumns()) {
+      for (ColumnDescriptor columnDescriptor : requestedSchema.getColumns()) {
+        if (columnChunkMetaData.getPath().equals(ColumnPath.get(columnDescriptor.getPath())) &&
+            isOnlyDictionaryEncodingPages(columnChunkMetaData.getEncodings())) {
+          long startingPosition = columnChunkMetaData.getStartingPos();
+          FSDataInputStream inputStream = path.getFileSystem(configuration).open(path);
+          inputStream.seek(startingPosition);
+
+          int totalSize = (int) columnChunkMetaData.getTotalSize();
+          byte[] buffer = new byte[totalSize];
+          inputStream.readFully(buffer);
+
+          DictionaryStream dictionaryStream = new DictionaryStream(buffer, 0);
+          DictionaryPage compressedPage = dictionaryStream.readDictionaryPage();
+          inputStream.close();
+
+          if (compressedPage == null) {
+            dictionaries.put(columnDescriptor, null);
+            break;
+          }
+
+          CodecFactory codecFactory = new CodecFactory(configuration);
+          BytesDecompressor decompressor = codecFactory.getDecompressor(columnChunkMetaData.getCodec());
+          DictionaryPage dictionaryPage = new DictionaryPage(decompressor.decompress(compressedPage.getBytes(),
+                                                                                    compressedPage.getUncompressedSize()),
+                                                            compressedPage.getDictionarySize(),
+                                                            compressedPage.getEncoding());
+          Dictionary dictionary = dictionaryPage.getEncoding().initDictionary(columnDescriptor, dictionaryPage);
+          dictionaries.put(columnDescriptor, dictionary);
+          break;
+        }
+      }
+    }
+    return dictionaries;
+  }
```

Comment:
Let's not make this a static method. We are duplicating things like the CODEC initialization in here.
I would expect to have something similar to the existing APIs (readNextRowGroup() but just for dictionaries):
```
DictionaryPageReadStore dicts = new ParquetFileReader(conf, fileMetadata, path, blocks, columns).readNextRowGroupDictionaries();
```
The decoding should not happen in this class.
Possibly you can add a method in MessageColumnIO to return a dictionary reader since it is the class responsible to create RecordReaders (see MessageColumnIO.getRecordReader and ColumnReadStoreImpl)